src/timing.h

```diff
@@ -300,33 +302,35 @@ STATIC_INLINE void _jl_timing_counts_stop(jl_timing_counts_t *block, uint64_t t)
     assert(block->running);
     block->running = 0;
 #endif
-    block->total += t - block->t0;
+    block->total += t - block->start;
 }
 
 STATIC_INLINE void _jl_timing_counts_start(jl_timing_counts_t *block, uint64_t t) JL_NOTSAFEPOINT {
 #ifdef JL_DEBUG_BUILD
     assert(!block->running);
     block->running = 1;
 #endif
-    block->t0 = t;
+    block->start = t;
 }
 
-STATIC_INLINE void _jl_timing_counts_ctor(jl_timing_counts_t *block) JL_NOTSAFEPOINT {
+STATIC_INLINE void _jl_timing_counts_ctor(jl_timing_counts_t *block, uint64_t t) JL_NOTSAFEPOINT {
     block->total = 0;
+    block->t0 = t;
```

Comment:
It needs to be in the constructor because we start and stop a block whenever we have a nested block, but we only construct it once. We shouldn't be relying on the time difference between construction and starting or else we risk not accounting for some time when profiling.

Reply:
I mean to put this in `jl_timing_block_start` which is only called once per block